loki/bulk/scheduler.py

```diff
@@ -222,7 +228,26 @@ def _discover(self):
         self.obj_map.update(
             (f'{module.name}#{r.name}', obj)
             for obj in obj_list for module in obj.modules
-            for r in module.subroutines + tuple(module.typedefs.values())
+            for r in module.subroutines + tuple(module.typedefs.values()) + module.variables
+        )
+        self.obj_map.update(
+            (f'{module.name}#{r.spec.name}', obj)
+            for obj in obj_list for module in obj.modules
+            for r in module.interfaces if r.spec
+        )
+
+        # Create maps of routines and module variables to help with fast sorting in create_item later
```

Comment:
Not convinced that performance justifies the bookkeeping overhead for additional maps here: Finding this out after a lookup in obj_map should be fairly quick, especially compared to the cost of transformations? Happy to be convinced otherwise, though!

Reply:
Yes agreed, those extra maps felt a bit clunky looking back at it and I've gotten rid of them. Thanks for the tip!